app/models/concerns/links_generator.rb

```diff
@@ -0,0 +1,24 @@
+module LinksGenerator
+  extend ActiveSupport::Concern
+
+  def new_link?
+    find_url_clean_duplicate.nil?
+  end
+
+  def find_url_clean_duplicate
+    Url.find_by(url_clean: url_clean)
+  end
+
+  def clean_url
+    clean_url = url_orginal.strip
+    clean_url = clean_url.downcase.gsub(%r{(https?:\/\/)|(www\.)}, '')
+    clean_url.slice!(-1) if clean_url[-1] == '/'
+    self.url_clean = clean_url
```

Comment:
Concerns and modules that rely on the presence of some attributes or external methods indicate that those aren't the appropriate abstractions for this problem. Ideally concerns / modules should be independent on the class they are included in, otherwise it's just a fancy way to split the code between different files